zerver/lib/actions.py

```diff
@@ -2242,7 +2240,9 @@ def check_message(sender: UserProfile, client: Client, addressee: Addressee,
         elif stream_id is not None:
             stream = validate_stream_id_with_pm_notification(stream_id, realm, sender)
         else:
-            stream = addressee.stream()
+            addressee_stream = addressee.stream()
+            assert addressee_stream is not None
+            stream = addressee_stream
```

Comment:
This seems to make `zerver.tests.test_audit_log.TestRealmAuditLog.test_user_activation` fail

Reply:
I'll look at the test code, but I'm surprised since this would seem to imply that it's OK to have an addressee which *is* a stream but can't provide a stream? Hmm, actually it's not running the tests, perhaps since this is one of those new-fangled 'draft' pull requests.

Reply:
Apparently the tests have run, but the result is due to the default `stream = None` being removed, and I didn't notice that this function returns that data in the dict, even if `None`. I'm not sure if that's intended as I've not checked where it's used, but I've updated this to reinstate setting it, but in the private branch with a comment instead.